src/features/utils/instrument-base.js

```diff
@@ -43,46 +52,53 @@ export class InstrumentBase extends FeatureBase {
    * @param {Object} [argsObjFromInstrument] - any values or references to pass down to aggregate
    * @returns void
    */
-  importAggregator (argsObjFromInstrument) {
-    if (this.hasAggregator || !this.auto) return
-    this.hasAggregator = true
-    let session, agentSessionImport
-    if (getConfigurationValue(this.agentIdentifier, 'privacy.cookies_enabled') === true && !isWorkerScope) {
-      agentSessionImport = import(/* webpackChunkName: "session-manager" */ './agent-session')
-        .catch(err => {
-          warn('failed to import the session manager', err)
-        })
-    }
+  importAggregator (argsObjFromInstrument = {}) {
+    if (this.featAggregate || !this.auto) return
+    const enableSessionTracking = isBrowserScope && getConfigurationValue(this.agentIdentifier, 'privacy.cookies_enabled') === true
+    let loadedSuccessfully, loadFailed
+    this.onAggregateImported = new Promise((resolve, reject) => {
+      loadedSuccessfully = resolve; loadFailed = reject
+    })
+
     const importLater = async () => {
+      try {
+        if (enableSessionTracking) { // would require some setup before certain features start
+          const { setupAgentSession } = await import(/* webpackChunkName: "session-manager" */ './agent-session')
+          const sessionManager = setupAgentSession(this.agentIdentifier)
+
+          argsObjFromInstrument.sessionTrackingOn = true // let relevant features know the manager was initialized successfully and should be found on runtime obj
+
+          if (this.featureName === FEATURE_NAMES.sessionReplay) {
+            if (!shouldImportSR(sessionManager)) {
+              drain(this.agentIdentifier, this.featureName)
+              return
+            }
+          }
+        }
```

Comment:
Need an else block to prevent the session replay feature from starting when sessions are disabled.

```suggestion
        } else if (this.featureName === FEATURE_NAMES.sessionReplay) {
          drain(this.agentIdentifier, this.featureName)
          return
        }
```